src/features/query-bar/components/filter/FilterButton.js

```diff
@@ -56,12 +56,11 @@ class FilterButton extends React.Component<Props, State> {
         });
     }
 
-    componentDidUpdate() {
-        const { conditions } = this.props;
-        const { transientConditions } = this.state;
-        const wasFlyoutClosed = transientConditions.length === 0;
-        const shouldHydrateConditions = transientConditions.length !== conditions.length;
-        if (wasFlyoutClosed && shouldHydrateConditions) {
+    componentDidUpdate(prevProps: Props, prevState: State) {
+        const { isMenuOpen, transientConditions } = this.state;
+        const { isMenuOpen: prevIsMenuOpen } = prevState;
+        const wasApplyClicked = transientConditions.length === 0 && isMenuOpen && !prevIsMenuOpen;
```

Comment:
I'm curious what behaves differently if you omit "transientConditions.length === 0"

Reply:
Here is the behavior when the above is removed:

On initial load, No Filters Applied is being shown (due to Redux having no initialized conditions).